### PR TITLE
OAK-11324 : removed usage of Guava's Sets.symmetricDifference

### PR DESCRIPTION
**Pull request by rishabhdaim**

### CODE REVIEW

File: oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java [hunk 1]
```diff
@@ -308,6 +308,68 @@ public void testIntersectionWithNoCommonElements() {
...
+        final Set<String> result = CollectionUtils.symmetricDifference(set1, set2);
+
+        final Set<String> expected = Set.of("a", "d");
+        Assert.assertEquals(expected, result);
+    }
+
+    @Test
+    public void testSymmetricDifferenceWithEmptySet() {
+        final Set<String> set1 = Set.of("a", "b", "c");
+        final Set<String> set2 = new HashSet<>();
```
Comment (reschke) [inline oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java]:
```suggestion
        final Set<String> set2 = Set.of();
```

File: oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java [hunk 2]
```diff
@@ -308,6 +308,68 @@ public void testIntersectionWithNoCommonElements() {
...
+
+        final Set<String> expected = Set.of("a", "b", "c");
+        Assert.assertEquals(expected, result);
+    }
+
+    @Test
+    public void testSymmetricDifferenceWithBothEmptySets() {
+        final Set<String> set1 = new HashSet<>();
+        final Set<String> set2 = new HashSet<>();
+
```
Comment (reschke) [inline oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java:337]: Same here.

**Review by reschke — APPROVED** [review]
...with minor comments.

File: oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java (hunk 1 above)
Comment (rishabhdaim) [reply oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java]: done in https://github.com/apache/jackrabbit-oak/pull/1924/commits/78590cb471e03f626fd91f904dd507171759090b

File: oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java (hunk 2 above)
Comment (rishabhdaim) [reply oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java:337]: done in https://github.com/apache/jackrabbit-oak/pull/1924/commits/78590cb471e03f626fd91f904dd507171759090b

**Pull request merged**